### ISSUE TITLE
'pip' uses py2.7 instead of py3 for core GeoNode install 

### ISSUE THREAD

**Issue by gannebamm** [GeoNode/geonode#5597]
## Expected Behavior
Following current [master branch docs](https://github.com/GeoNode/geonode/blob/master/docs/install/core/index.rst) I should be able to install core geonode:

## Actual Behavior
After trying to [execute pip tasks](http://docs.geonode.org/en/master/install/core/index.html#geonode-installation):
```
pip install -r requirements.txt --upgrade --no-cache --no-cache-dir
pip install -e . --upgrade
```
I get errors because `pip` is linked to python 2.7 on my ubuntu server 18.04 LTS. 

## Steps to Reproduce the Problem

  1. follow the instructions at [master branch docs](https://github.com/GeoNode/geonode/blob/master/docs/install/core/index.rst)

## Specifications

  - GeoNode version: GeoNode 3.0.dev1580801344
  - Installation method: manual
  - Platform: Ubuntu Server 18.04 LTS
  - Additional details:
      Using `pip3` solves the issue for me

**Comment by gannebamm** [comment]
After redoing the docs with a fresh machine I did not stumble across this issue anymore. Maybe there was some old configuration on my old VM causing this issue?

If no one else faces this issue I will close it.

**Comment by giohappy** [comment]
@gannebamm this is sonething related to the specific environments (which versiin of python pip uses) and not to GeoNode. Can this issue be closed?

**Comment by gannebamm** [comment]
@giohappy Even though you are right, following the master doc from an ubuntu 18.04 server should be working without hassle. Nonetheless I close this now

**Comment by travislbrundage** [comment]
Just to state an alternative, I installed `python3.7-dev` on my Ubuntu LTS 18.04 machine and then did `python3.7 -m pip` for the pip install commands and that solved this problem for me, since as you mentioned, the default is python 3.6.

**Comment by albertStaalburger** [comment]
I just got that error, fresh Ubuntu 18.04.
I needed to install a barrage of other packets and after some back and forth got it working.
Packets I had to install:
- sudo apt-get install -y gcc
- sudo apt-get install libpq-dev python-dev python3-dev

I initially only installed python-dev, but had no luck, after installing python3-dev things seemd to come together.

I plan on nuking the machine and trying again.

**Comment by afabiani** [comment]
@albertStaalburger which instructions are you following exactly?

For GeoNode 3.0 you should follow docs 3.0 or master

As an instance:

https://docs.geonode.org/en/master/install/core/index.html#packages-installation

I see in this page the following instructions:

```
...
sudo apt install -y python3-pip python3-dev python3-virtualenv python3-venv virtualenvwrapper
...
sudo apt install -y git unzip gcc zlib1g-dev libgeos-dev libproj-dev
...
```

Is there something missing on those packages? Should we add `libpq-dev` too?

**Comment by albertStaalburger** [comment]
I am following instructions on `https://docs.geonode.org/en/master/install/project/index.html`
The docs indicate `master`.

When trying to fix the installation the first time I followed the error messages and installed the missing packages.  The `libpq-dev` package popped up on SO as beign a required package for some of the python processes.

After canning the server and starting over I did not get the error again, and as such did not need to install any extra packages beyond the ones already listed.
I assume that with the first attempt some of the packages weren't installed.
